### PR TITLE
Imp/html template

### PR DESCRIPTION
**Pull request by patricksimonian**
allow for HTML templates to be created. This feature has been flagged off but its proof of concept is working.

### CODE REVIEW

File: app-web/gatsby/createPages.js [hunk 1]
```diff
@@ -53,14 +59,19 @@ module.exports = async ({ graphql, boundActionCreators }) => {
...
+    const isResource =
+      node.childMarkdownRemark &&
+      node.childMarkdownRemark.frontmatter &&
+      node.childMarkdownRemark.frontmatter.resourcePath;
+    const isIgnored =
+      node.childMarkdownRemark &&
+      node.childMarkdownRemark.frontmatter &&
+      node.childMarkdownRemark.frontmatter.ignore;
+    // if file is html these would both resolve to false since there are no meta data properties
+    if (!isResource && !isIgnored && node.internal.mediaType === 'text/markdown') {
```
Comment (ShellyXueHan) [inline app-web/gatsby/createPages.js]: add comment here

File: app-web/plugins/gatsby-source-github-all/sourceNodes.js [hunk 2]
```diff
@@ -38,6 +38,8 @@ const createGHNode = (file, id) => ({
   sourceName: file.metadata.sourceName, // the pretty name of the 'source'
   sourcePath: file.metadata.sourceURL, // the path to the repo
   resourcePath: file.metadata.resourcePath, // either path to a gastby created page based on this node
+  resourceTitle: file.metadata.resourceTitle,
+  resourceDescription: file.metadata.resourceDescription,
```
Comment (ShellyXueHan) [inline app-web/plugins/gatsby-source-github-all/sourceNodes.js]: Are you planning to use the description somewhere?

File: app-web/plugins/gatsby-source-github-all/utils/plugins.js [hunk 3]
```diff
@@ -84,30 +90,31 @@ const markdownFrontmatterPlugin = (extension, file) => {
...
+      file.metadata.resourcePath = frontmatter.resourcePath;
+      return file;
+    }
   }
-
+  // no resource path, this file is destined to be turned into a page,
+  // the page page is composed of the source name, the title of the file plus an id
+  file.metadata.resourcePath = createPathWithDigest(
+    file.metadata.source,
+    file.metadata.source,
```
Comment (ShellyXueHan) [inline app-web/plugins/gatsby-source-github-all/utils/plugins.js]: duplication?

File: app-web/src/templates/SourceMarkdown.module.css [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+.MarkdownBody h5,
+.MarkdownBody h6 {
+    color: #6d6d6d;
+    font-weight: 400;
+}
+
+.MarkdownBody p {
+    line-height: 1.25em;
+    color: #3e3e3e;
+    font-size: 14x;
```
Comment (ShellyXueHan) [inline app-web/src/templates/SourceMarkdown.module.css]: px

File: app-web/src/hoc/TemplateLayout.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                <a href={`${siphonData.sourcePath}fork`}>
+                  <FontAwesomeIcon
+                    icon={faCodeBranch}
+                    aria-label={`Fork ${siphonData.sourceName} on github`}
+                  />
+                </a>
+              </li>
+            </ul>
+          </header>
+          {/* navigation doesn't show up if there are no links */}
```
Comment (ShellyXueHan) [inline app-web/src/hoc/TemplateLayout.js]: better move this comment to line 13 so it's clear

File: app-web/plugins/gatsby-source-github-all/utils/plugins.js (hunk 3 above)
Comment (patricksimonian) [reply app-web/plugins/gatsby-source-github-all/utils/plugins.js]: thats not a duplication, this fn is called `createPathWithDigest(baseURL, ...digestables)`
should I comment that?

File: app-web/plugins/gatsby-source-github-all/sourceNodes.js (hunk 2 above)
Comment (patricksimonian) [reply app-web/plugins/gatsby-source-github-all/sourceNodes.js]: yes, this is taking one step to an isolated siphon node that is not dependant on what the source type is. This is in prep for the unfurling PR

**Pull request merged**